src/packet/tcp.rs.in

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2014, 2015 Robert Clipsham <robert@octarineparrot.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use packet::Packet;
+use packet::PrimitiveValues;
+use packet::ip::IpNextHeaderProtocol;
+
+use pnet_macros_support::types::*;
+
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+/// Represents a TCP Packet
+#[packet]
+pub struct Tcp {
+    source: u16be,
+    destination: u16be,
+    seq: u32be,
+    ack_seq: u32be,
+    data_offset: u4,
+    reserved: u4,
+    cwr: u1,
+    ece: u1,
+    urg: u1,
+    ack: u1,
+    psh: u1,
+    rst: u1,
+    syn: u1,
+    fin: u1,
```

Comment:
I'm kinda torn here. While it's nice to have these as separate fields, it makes setting them all at once difficult. I think my preferred approach is to have a single `flags` field, then have an enumeration for each value. I can be persuaded either way though, I'd be interested to hear your thoughts.